pkg/tsdb/cloudwatch/query_transformer.go

```diff
@@ -93,10 +91,10 @@ func (e *cloudWatchExecutor) transformQueryResponsesToQueryResult(cloudwatchResp
 		})
 
 		if requestExceededMaxLimit {
-			queryResult.ErrorString = "Cloudwatch GetMetricData error: Maximum number of allowed metrics exceeded. Your search may have been limited."
+			queryResult.Error = fmt.Errorf("cloudwatch GetMetricData error: Maximum number of allowed metrics exceeded. Your search may have been limited")
```

Comment:
The reason ErrorString and not Error was used here is because we want to display an error on the top left of the panel to indicate something went wrong, but at the same the we'd like the time series that we got from the API to be displayed in the panel. By using Error, nothing will be rendered in the panel - we will only get an error message. 

So really this is just a warning. When this was implemented about a year ago, the api didn't have any support to propagate warnings to the panel, but maybe there is a way of doing that today? 

This is pretty hard to reproduce, because in case this limit is reached, the partial data limit is also reached and in that case, the partial data error message will be displayed.